src/k8s/pkg/component/component.go

```diff
@@ -67,6 +79,47 @@ func (h *helmClient) Enable(componentName string, values map[string]any) error {
 		return fmt.Errorf("failed to initialize Helm client configuration: %w", err)
 	}
 
+	// Wait for any functionality that this one depends on.
+	// It will not enable that parent functionality.
+	if component.DependsOn != "" {
+		parent, ok := h.components[component.DependsOn]
+		if !ok {
+			return fmt.Errorf("invalid parent %s for %s", componentName, component.DependsOn)
+		}
+		parentConfig, err := h.initializer.New(parent.Namespace)
+		if err != nil {
+			return fmt.Errorf("failed to initialize Helm client configuration: %w", err)
+		}
+
+		get := action.NewGet(parentConfig)
+		release, err := get.Run(parent.ReleaseName)
+		if err != nil {
+			return fmt.Errorf("failed to get release %s: %w", parent.ReleaseName, err)
+		}
+		resources, err := parentConfig.KubeClient.Build(bytes.NewBufferString(release.Manifest), false)
+		if err != nil {
+			return fmt.Errorf("failed to build kubernetes objects from release manifest: %w", err)
+		}
+		timeout := 30 * time.Second
+		if deadline, set := ctx.Deadline(); set {
+			timeout = time.Until(deadline)
+		}
+		err = parentConfig.KubeClient.Wait(resources, timeout)
+		if err != nil {
+			fmt.Println(err)
+			return err
+		}
+
+		// TODO: This should not be necessary but it seems that the implementation of helms "Wait" command does
+		// not properly wait for the Pods to come up.
+		for _, resource := range resources {
+			cmd := []string{"k8s", "kubectl", "wait", "--for=condition=ready", "pod", "--all", "-n", resource.Namespace, "--timeout", "30m"}
+			// Ignore any error that pops up.
+			// An expected error is e.g. "resource not found" for namespaces that do not have pods.
+			utils.RunCommand(ctx, cmd...)
```

Comment:
this should not be needed, and the `utils.RunCommand()` is hard to write unit tests with. What happens if we remove it?